opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -374,18 +385,21 @@ def collect(self) -> None:
         self._collect_observers()
 
     def _collect_metrics(self) -> None:
-        for metric in self.metrics:
-            if not metric.enabled:
+        for instrument in self.instruments.values():
```

Comment:
With this we will be iterating `self.instruments` twice? Can we not just loop through it once and do type checking for each, collecting each appropriately depending on whether it's a metric or an observer?

Reply:
Yes, we can do that.